### DISCUSSION TITLE
Как сделать в таблице, чтобы у дней недели чередовались цвета по-очереди

### DISCUSSION THREAD

**Discussion by gorbarov** [totumonline/totum-ru-issues-and-discussions#357] [Как это сделать?] [answered]
Привет, появилась у меня задачка не совсем тривиальная, но интересная. 

Есть таблица, в которой собираются приходы и расходы за период.
По реализации это временная таблица, сначала список собирается в h_header_input

Я хочу, чтобы соседние дни были разного цвета. Но есть пара но:
1) Дни могут идти не по порядку, поэтому идея с чётностью чисел отпадает, т.к. может быть 1, 3, 5 число.
2) В один день может быть много платежей и все они должны быть одного цвета, чтобы чередовались именно дни месяца, а не строки.

Я такое делал на google script, скрин ниже. Но там это в цикле сделано, в котором переменная сбрасывается по-очереди в true/false когда следующий день в цикле перебора случается.

Как сделать такое в Totume я что-то ума не приложу, но с такой штукой сильно удобнее работать, информация легче считывается.

![2023-04-28_09-52-35](https://user-images.githubusercontent.com/388620/235043397-27c61ede-68bd-4a2f-bb1d-1b57209ec39e.png)

**Comment by poluninatm** [comment]
Не, там не про типовую черезстрочность - там в дне может быть несколько
строк, а краситься они должны одинаково.

Если не порешаете, я сегодня как до компа доберусь - оформлю мое решение:
сейчас думаю, красиво будет сделать через юник по дням с сортировкой и
четность индекса этого массива.

Пт, 28 апр. 2023 г. в 08:48, erpic-ru ***@***.***>:

**Reply by gorbarov** [reply, marked as answer]
Да, коллега подсказал уже! Отлично работает

```
=: $currentRow[0]['style']

currentRow: listFilter(list: $listStyled; key: "item" = #data; item: "data")
	listStyled: listReplace(list: $listSorted; action: $action; key: "key"; value: "value")
		action: rowCreate(field: "data" = $#value; field: "style" = math`$round - $division`; field: "key" = $#key)
			round: round(num: $division; type: "up"; step: 1; dectimal: 0)
				division: $#key / 2
	listSorted: listSort(list: $listUniq; type: "regular"; direction: "asc";)	
		listUniq: listUniq(list: $listDates)
			listDates: selectList(table: $#ntn; field: 'data')
```

И в условном форматировании ячейки

`f1=: setrowformat(condition: $#nfv != "0"; background: "LightGray")`

А можно как-то всю строку закрасить? Или только для кажой ячейки применять стиль?

**Reply by gorbarov** [reply]
Так вроде сами предложили оформить своё решение...

**Comment by poluninatm** [comment]
SetRowFormat используйте в коде форматирования строки. И он покрасит всю
строку, если ячейка себя не перекрасит

В коде форматирования поля - setFormat.

Пт, 28 апр. 2023 г. в 09:18, gorbarov ***@***.***>:

**Comment by gorbarov** [comment]
@ttm-alexeypolunin Алексей, если будет время, я был бы рад вашему прочтению такого решения.

Наше на 350 строках грузится 30 секунд )))

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/388620/123d3676-280c-41a9-99d6-92ab37249b07)

Актуальный код такой:
```
=: $currentRow[0]['style']

currentRow: listFilter(list: $listStyled; key: "item" = #data; item: "data")
	listStyled: listReplace(list: $listSorted; action: $action; key: "key"; value: "value")
		action: rowCreate(field: "data" = $#value; field: "style" = math`$round - $division`; field: "key" = $#key)
			round: round(num: $division; type: "up"; step: 1; dectimal: 0)
				division: $#key / 2
	~listSorted: listSort(list: $listUniq; type: "regular"; direction: "asc";)	
		~listUniq: listUniq(list: $listDates)
			~listDates: selectList(table: $#ntn; field: 'data')
```

**Reply by ttm-alexeypolunin** [reply]
https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/20#discussioncomment-3535961

**Reply by gorbarov** [reply]
Вы же сами, чуть выше в переписке предложили оформить своё решение. Я не настаиваю.
